### PR TITLE
Support using buttons and axis in the same message

### PR DESCRIPTION
**Pull request by Timple**
Eventhough a button can have only two values, it would be nice if they could still be used next to axes in the same message.

### CODE REVIEW

**Comment by Timple** [comment]
Mild bump :)

I do not expect this PR to break current implementations. Does something still need to be modified?

File: joy_teleop/scripts/joy_teleop.py [hunk 1]
```diff
@@ -188,14 +188,24 @@ def run_topic(self, c, joy_state):
...
         else:
             for mapping in cmd['axis_mappings']:
-                if len(joy_state.axes)<=mapping['axis']:
-                  rospy.logerr('Joystick has only {} axes (indexed from 0), but #{} was referenced in config.'.format(len(joy_state.axes), mapping['axis']))
-                  val = 0.0
+                if 'axis' in mapping:
+                    if len(joy_state.axes) <= mapping['axis']:
+                        rospy.logerr('Joystick has only {} axes (indexed from 0), but #{} was referenced in config.'.format(len(joy_state.axes), mapping['axis']))
+                        val = 0.0
+                    else:
```
Comment (bmagyar) [inline joy_teleop/scripts/joy_teleop.py]: could you flip the logic here please? 

it is more intuitive to do it with the cases you intend to cover first, then error out on the rest

File: joy_teleop/scripts/joy_teleop.py [hunk 2]
```diff
@@ -188,14 +188,24 @@ def run_topic(self, c, joy_state):
...
+                        val = joy_state.axes[mapping['axis']] * mapping.get('scale', 1.0) + mapping.get('offset', 0.0)
+                elif 'button' in mapping:
+                    if len(joy_state.buttons) <= mapping['button']:
+                        rospy.logerr('Joystick has only {} buttons (indexed from 0), but #{} was referenced in config.'.format(len(joy_state.axes), mapping['button']))
+                        val = 0.0
+                    else:
```
Comment (bmagyar) [inline joy_teleop/scripts/joy_teleop.py]: Also flip it here. Do the processing case first, then the error handling

**Comment by bmagyar** [comment]
Could you please also add an update to the example file [here](https://github.com/ros-teleop/teleop_tools/blob/indigo-devel/joy_teleop/config/joy_teleop_example.yaml)?

**Comment by Timple** [comment]
Let me know if you need a rebase!

**Comment by bmagyar** [comment]
Thanks for the contribution and sticking around :)

**Pull request merged**